omod/src/main/java/org/openmrs/module/appointments/web/controller/AppointmentController.java

```diff
@@ -60,6 +60,10 @@ public List<AppointmentDefaultResponse> getAllAppointments(@RequestParam(value =
     @RequestMapping( method = RequestMethod.POST, value = "search")
     @ResponseBody
     public List<AppointmentDefaultResponse> searchAppointments( @Valid @RequestBody AppointmentQuery searchQuery) throws IOException {
+        if(searchQuery.getIsDatelessAppointments()) {
```

Comment:
Also you are expecting a searchquery but fetching only date-less appointment? 
Does it mean 
- if appointment without dates, rest of criteria do not apply? That does not seem right. I can look for an appointment with a certain priority or service for which dates have not been assinged yet - right?

please don't call service method as "searchDatelessAppointment".  Call it "appointmentWithoutDatesAssigned" or something that makes more sense. DatelessAppt seems very much a technical internal thing.

Reply:
renamed getter as isWithoutDates() , service method as searchAppointmentsWithoutDates() and dao method as getAppointmentsWithoutDates()

Reply:
same question above is relevant here as well. Why not leverage status if you are looking for "waitlisted" appointments? Unless you want to return all appointments without dates?